lib/kubernetes-deploy/task_config.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class TaskConfig
+    attr_reader :context, :namespace
+
+    def initialize(context, namespace, logger = nil)
+      @context = context
+      @namespace = namespace
+      @logger = logger
+    end
+
+    def logger
+      @logger ||= KubernetesDeploy::FormattedLogger.build(@namespace, @context)
+    end
+
+    def kubectl(log_failure_by_default: true)
```

Comment:
What is the reason for adding kubectl/kubeclient in here? They themselves aren't configuration--they're really powerful cluster access objects. `TaskConfig` strikes me as something we could pass around to all our business objects instead of passing context, namespace and logger individually, but many of those are not supposed to talk to the cluster.